lottie/src/main/java/com/airbnb/lottie/LottieAnimationView.java

```diff
@@ -806,6 +809,24 @@ private void clearComposition() {
     lottieDrawable.clearComposition();
   }
 
+  /**
+   * If rendering via software, Android will fail to generate a bitmap if the view is too large. Rather than displaying
+   * nothing, fallback on hardware acceleration which may incur a performance hit. minSdkVersion is 16. If this is ever
+   * lowered we need to add a check `Build.VERSION.SDK_INT < Build.VERSION_CODES.HONEYCOMB` to determine if hardware
+   * acceleration is supported.
+   *
+   * @see #setRenderMode(RenderMode)
+   * @see com.airbnb.lottie.LottieDrawable#draw(android.graphics.Canvas)
+   */
+  @Override
+  public void buildDrawingCache(boolean autoScale) {
+    super.buildDrawingCache(autoScale);
+    if (getLayerType() == LAYER_TYPE_SOFTWARE && getDrawingCache(autoScale) == null) {
+      setRenderMode(HARDWARE);
+      requestLayout();
```

Comment:
Why requestLayout()?

Reply:
hmm good point. i was thinking that since this is called during draw, we would need to retrigger it but the logic there would fall into hardware stuff since software is first. removing